1_slices_modules_and_objects/exercises/binary_search/src/lib.rs

```diff
@@ -1,7 +1,27 @@
 /// Finds the index of `value` in the sorted slice `v`
 /// Returns the index if found, None otherwise
 pub fn binary_search(v: &[i32], value: i32) -> Option<usize> {
-    unimplemented!();
+    let mut high: i32 = v.len() as i32 - 1;
+    let mut low = 0;
+    
+    while high >= low {
+        let mid = ((high - low)) / 2 + low;
+        let mid_index = mid as usize;
+        let cur = v[mid_index];
+
+        if cur == value {
+            return Some(mid_index);
+        }
+        if cur < value {
+            low = mid + 1;
+        } else {
+            high = mid - 1;
+        }
+    }
+
+    None
+
+    // Is it legit to allow overflow here since we know it will not cause issue ?
```

Comment:
Which overflow? Also, all your operations are checked and will panic at runtime in debug mode if they overflow. However, overflows won't be checked in release builds (`cargo run --release` or `cargo build --release`) or I'd you use unchecked operations (you can search "unchecked addition rust" on Google to see how to proceed about that)

Reply:
I mean could we explicitly allow overflow in debug as well in this case using unsafe since we are 100% sure it will not cause issue ?
(would have used it for the high var since if we check overflow, it will when the value will not be found at the left most index with mid - 1).
An other way to fix this issue might be to use high = mid as a workaround but could lead to less optimized code)

Reply:
For sure! Did you check out how to do that? Also if your main concern is performance, don't worry about it. Debug builds are not optimized and have a looooool of runtime checks, so it's not worth bypassing one of Rust's security for extra performance in debug mode. And in release mode, the check will be removed